### PR TITLE
Support appsettings.json as a config file for rocklib

### PR DESCRIPTION
**Pull request by JustinMoss**

### CODE REVIEW

**Comment by derikwhittaker** [comment]
Simple enough.

Not a fan of IF blocks without {} but that is my style of choice.  Not a deal breaker

File: RockLib.Configuration.AppSettingsConfigTests/ConfigTests.cs [hunk 1]
```diff
@@ -48,10 +50,16 @@ public void EnvironmentVariablesDoNotOverrideAfterLocked()
...
             // Setting it this time, however, is too late.
-            Environment.SetEnvironmentVariable("AppSettings:Environment", "Beta");
+            Environment.SetEnvironmentVariable("AppSettings:RockLib.Environment", "Beta");
 
             // No effect.
-            Assert.Equal("Prod", Config.AppSettings["Environment"]);
+            Assert.Equal("Prod", Config.AppSettings["RockLib.Environment"]);
+        }
+
+        [Fact(DisplayName = "AppSettingsConfigTests: Changing environment variable values when IsLocked is true has no effect.")]
```
Comment (bfriesen) [inline RockLib.Configuration.AppSettingsConfigTests/ConfigTests.cs]: DisplayName looks like a copypasta error.

File: RockLib.Configuration.AppSettingsConfigTests/ConfigTests.cs [hunk 2]
```diff
@@ -28,7 +30,7 @@ public void EnvironmentVariablesOverrideFileConfig()
         {
             // Note that the "AppSettings:Environment" environment variable was
```
Comment (bfriesen) [inline RockLib.Configuration.AppSettingsConfigTests/ConfigTests.cs]: Should probably look for "Environment" (without the "RockLib." prefix) in comments and replace with prefixed.

File: RockLib.Configuration/RockLibConfigurationBuilderExtensions.cs [hunk 3]
```diff
@@ -38,7 +42,7 @@ public static IConfigurationBuilder AddRockLib(this IConfigurationBuilder builde
         /// <exception cref="ArgumentNullException">If <paramref name="builder"/> is null.</exception>
         /// <exception cref="ArgumentException">If <paramref name="jsonConfigPath"/> is null or empty.</exception>
         /// <returns>The <see cref="IConfigurationBuilder"/>.</returns>
-        public static IConfigurationBuilder AddRockLib(this IConfigurationBuilder builder, string jsonConfigPath, string supplementalJsonConfigPath = null)
+        public static IConfigurationBuilder AddJson(this IConfigurationBuilder builder, string jsonConfigPath, string supplementalJsonConfigPath = null)
```
Comment (bfriesen) [inline RockLib.Configuration/RockLibConfigurationBuilderExtensions.cs]: Does this method provide any value anymore? Should it be made private?

**Review by bfriesen — CHANGES_REQUESTED** [review]
A couple of minor things and a question. See comments.

File: RockLib.Configuration/RockLibConfigurationBuilderExtensions.cs (hunk 3 above)
Comment (JustinMoss) [reply RockLib.Configuration/RockLibConfigurationBuilderExtensions.cs]: Not really.  I'll clean it up, thanks.

File: RockLib.Configuration.AppSettingsConfigTests/ConfigTests.cs (hunk 1 above)
Comment (JustinMoss) [reply RockLib.Configuration.AppSettingsConfigTests/ConfigTests.cs]: Whoops.

File: RockLib.Configuration.AppSettingsConfigTests/ConfigTests.cs (hunk 2 above)
Comment (JustinMoss) [reply RockLib.Configuration.AppSettingsConfigTests/ConfigTests.cs]: Comments... get me every time :-/

**Review by bfriesen — APPROVED** [review]

File: RockLib.Configuration/RockLibConfigurationBuilderExtensions.cs [hunk 4]
```diff
@@ -11,46 +11,27 @@ namespace RockLib.Configuration
...
-        /// <param name="jsonConfigPath">The path of the json config file.</param>
-        /// <param name="supplementalJsonConfigPath">The path of the supplemental json config file.</param>
-        /// <exception cref="ArgumentNullException">If <paramref name="builder"/> is null.</exception>
-        /// <exception cref="ArgumentException">If <paramref name="jsonConfigPath"/> is null or empty.</exception>
-        /// <returns>The <see cref="IConfigurationBuilder"/>.</returns>
-        public static IConfigurationBuilder AddRockLib(this IConfigurationBuilder builder, string jsonConfigPath, string supplementalJsonConfigPath = null)
-        {
-            if (builder == null) throw new ArgumentNullException(nameof(builder));
-            if (string.IsNullOrEmpty(jsonConfigPath)) throw new ArgumentException($"'{nameof(jsonConfigPath)}' must be a non-empty string.", nameof(jsonConfigPath));
+            if (string.IsNullOrEmpty(environment))
```
Comment (derikwhittaker) [inline RockLib.Configuration/RockLibConfigurationBuilderExtensions.cs:29]: Seems like this could be an if/else rather than 2 ifs.  But nothing major

**Pull request merged**